homeassistant/components/light/zha.py

```diff
@@ -83,13 +110,81 @@ class Light(ZhaEntity, light.Light):
 
     _domain = light.DOMAIN
 
+    class OnOffListener:
```

Comment:
Why nest this in the entity class?

Reply:
Just copied how it was handled / accepted in binary_sensor. The next goal was to refactor out the common parts and share the logic. Is there something I should change?

Reply:
Flat is better than nested. We don't seem to need to nest it, so then we shouldn't.

Reply:
hows this?